### PR TITLE
Adding storages for each resource map

### PR DESCRIPTION
**Pull request by arschles**
This PR is a follow-on to #163, and requires merge of that PR before this can be considered for review. It adds the versions map to the API group. The versions map is a lookup table for the API server to match a versioned resource type to a storage mechanism.

This PR may require types to be generated.

~~**Requires https://github.com/kubernetes-incubator/service-catalog/pull/163 _before review_**~~

~~**Before this can be reasonably reviewed, this patch needs to be rebased against #173 after that PR is merged**~~ (#173 is now merged)

### CODE REVIEW

**Comment by MHBauer** [comment]
I've got some other spins on this I'd like to discuss. I also have a giant hack that we need to figure out how to get rid of. Good luck with the move.

File: pkg/apis/servicecatalog/v1alpha1/service_binding_storage.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package v1alpha1
+
+import (
```
Comment (pmorie) [inline pkg/apis/servicecatalog/v1alpha1/service_binding_storage.go]: we put the rest storage implementations elsewhere in k8s.  I'm in favor of keeping storage and API separate from a first-principles standpoint as well.  In k8s, the rest storage impl for api group package `foo` goes into `pkg/registry/foo`, with subpackages for each resource.  So if API group `foo` has resources `Bar` and `Fiz`, you would have:

```
pkg/
  registry/
    foo/
      bar/
      fiz/
```

Refactor to that?

Comment (arschles) [inline pkg/apis/servicecatalog/v1alpha1/service_binding_storage.go]: yep, will do

**Comment by arschles** [comment]
@MHBauer wanna discuss the spins today?

**Comment by arschles** [comment]
Closing in favor of #186

**Pull request closed without merging**